tests/unit/test_commons.h

```diff
@@ -99,45 +99,50 @@ using lsType = std::size_t;
 
 // SpliceTypeMeta: zip a set of types with ValueT
 
-template <int N, typename ValueT, typename TypeList>
-struct SpliceTypeMeta {
-  using ResultType = decltype(
-    std::tuple_cat(
-      std::declval<
-        std::tuple<
-          std::tuple<typename std::tuple_element<N-1,TypeList>::type,ValueT>
-        >
-      >(),
-      std::declval<typename SpliceTypeMeta<N-1,ValueT,TypeList>::ResultType>()
-    )
-  );
+template <int N, typename ValueT, typename... List>
+struct SpliceTypeMeta;
+
+template <int N, typename ValueT, typename Type, typename... List>
+struct SpliceTypeMeta<N, ValueT, Type, List...> {
+  using ResultType = std::tuple<
+    std::tuple<Type, ValueT>,
+    typename std::tuple_element<
+      0,
+      typename SpliceTypeMeta<N-1, ValueT, List>::ResultType
+    >::type...
+  >;
 };
 
-template <typename ValueT, typename TypeList>
-struct SpliceTypeMeta<0,ValueT,TypeList> {
-  using ResultType = decltype(std::declval<std::tuple<>>());
+template <typename ValueT, typename Type>
+struct SpliceTypeMeta<1, ValueT, Type> {
+  using ResultType = std::tuple<std::tuple<Type, ValueT>>;
 };
 
-template <typename TypeList, typename ValueT>
-struct SpliceMeta {
+template <typename ValueT, typename Tuple>
+struct SpliceMeta;
+
+template <typename ValueT, typename... Ts>
+struct SpliceMeta<ValueT, std::tuple<Ts...>> {
   using ResultTupleType = typename SpliceTypeMeta<
-    std::tuple_size<TypeList>::value,ValueT,TypeList
+    std::tuple_size<std::tuple<Ts...>>::value,
+    ValueT,
+    Ts...
   >::ResultType;
 };
 
-//
-// Debug code for SpliceTypeMeta
-//
+// //
```

Comment:
is this formatting for doc or accidental?

Reply:
Fixed